internal/cmd/base/flags.go

```diff
@@ -682,6 +682,112 @@ func mapToKV(m map[string]string) string {
 	return strings.Join(list, ",")
 }
 
+// StringSliceMapVar maps a key string to a slice of string values.
+// This is useful for cases such as modifying Worker tags, which can have
+// multiple values associated per key.
+type StringSliceMapVar struct {
+	Name       string
+	Aliases    []string
+	Usage      string
+	Default    map[string][]string
+	Hidden     bool
+	Target     *map[string][]string
+	Completion complete.Predictor
+}
+
+func (f *FlagSet) StringSliceMapVar(i *StringSliceMapVar) {
+	f.VarFlag(&VarFlag{
+		Name:       i.Name,
+		Aliases:    i.Aliases,
+		Usage:      i.Usage,
+		Value:      newStringSliceMapValue(i.Default, i.Target, i.Hidden),
+		Completion: i.Completion,
+	})
+}
+
+type stringSliceMapValue struct {
+	hidden bool
+	target *map[string][]string
+}
+
+func newStringSliceMapValue(def map[string][]string, target *map[string][]string, hidden bool) *stringSliceMapValue {
+	*target = def
+	return &stringSliceMapValue{
+		hidden: hidden,
+		target: target,
+	}
+}
+
+func (s *stringSliceMapValue) Set(val string) error {
+	kv := strings.TrimSpace(val)
+	if kv == "null" {
```

Comment:
This is sort of an in-between pattern, between the add/set/remove we have for e.g. grants on roles, and attributes/secrets like we have on host catalogs. The host catalogs have two flags: `attr` or `secret` for individual values, or `attributes` and `secrets` for setting the full value at once. Array fields like grants and principals on roles allow `null` only in a `set` call.

Based on that prior art I think we don't need a `tags` field right now, but I do think we should limit `null` to a `set` call, and track its usage to ensure that if `null` is passed during `set` that it's the only value passed.

Reply:
I understand limiting `set-worker-tags` to `null` only, based on the array fields precedent, but don't we need the `tags` field to allow `add-worker-tags` and `remove-worker-tags` to work?